frame/dvm/evm/src/runner/stack.rs

```diff
@@ -66,50 +67,59 @@ impl<T: Config> Runner<T> {
 		) -> (ExitReason, R),
 	{
 		let base_fee = T::FeeCalculator::min_gas_price();
-		// Gas price check is skipped when performing a gas estimation.
-		let max_fee_per_gas = match max_fee_per_gas {
-			Some(max_fee_per_gas) => {
-				ensure!(max_fee_per_gas >= base_fee, <Error<T>>::GasPriceTooLow);
+		let max_fee_per_gas = match (max_fee_per_gas, is_transactional) {
+			(Some(max_fee_per_gas), _) => {
+				ensure!(max_fee_per_gas >= base_fee, Error::<T>::GasPriceTooLow);
 				max_fee_per_gas
 			},
-			None => Default::default(),
+			// Use default gas price for internal transaction
```

Comment:
Why there is an internal transaction related comment in this generalized stack runner?

Request more explanation of this part before merging.

Reply:
For internal transactions, the tx sender is a special one and the gas price or (max_fee_per_gas, max_priority_fee_per_gas) is `None` https://github.com/darwinia-network/darwinia-common/blob/main/frame/dvm/ethereum/src/lib.rs#L821-L822. Besides, the internal transactions have the ability to change runtime storage, that means `is_transactional` is true in this case.

But, this will be seen as an invalid call in the frontier upstream runner validation, the `max_fee_per_gas` must be `Some` when `is_transactional` is true. Only if in the non-transactional scenario(estimate gas), the `max_fee_per_gas` can be `None`.

Reply:
I see, we proceed to review this again after s2s pallet get removed then.